### PR TITLE
The description field is not imported

### PR DESCRIPTION
**Pull request by votdev**
**Problem:**
The `Description` field of a compute server is not copied to the imported VM.

**Solution:**
Extract the `Description` field from the server data and set the `field.cattle.io/description` annotation for the imported VM.

**Related Issue:**
https://github.com/harvester/harvester/issues/6464

**Test plan:**
- Create a server in OpenStack and set a description.
- Import that VM into Harvester.
- The description should be set for the VM in Harvester.

Note, it seems, not every OpenStack installation is exposing the `Description` field via their REST API (Nova?) that is called by the `vm-import-controller`to fetch the server (VM) metadata.

### CODE REVIEW

File: pkg/source/openstack/client.go [hunk 1]
```diff
@@ -355,6 +364,9 @@ func (c *Client) GenerateVirtualMachine(vm *migration.VirtualMachineImport) (*ku
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      vm.Spec.VirtualMachineName,
 			Namespace: vm.Namespace,
+			Annotations: map[string]string{
```
Comment (ibrokethecloud) [inline pkg/source/openstack/client.go:367]: we should make this optional for when an actual description is returned, else we just end up with a key but no value.

rest looks good.

Comment (votdev) [reply pkg/source/openstack/client.go]: Done

**Review by ibrokethecloud — APPROVED** [review]
lgtm. thanks.

File: pkg/source/openstack/client.go [hunk 2]
```diff
@@ -361,6 +371,10 @@ func (c *Client) GenerateVirtualMachine(vm *migration.VirtualMachineImport) (*ku
 		},
 	}
 
+	if vmObj.Description != "" {
+		newVM.ObjectMeta.Annotations[annotationDescription] = vmObj.Description
```
Comment (starbops) [inline pkg/source/openstack/client.go:375]: Do you think we need to ensure the `Annotations` field is not `nil` in the first place?

Comment (votdev) [reply pkg/source/openstack/client.go:366]: You are right. `Annotations` are not initialized at this time.

**Review by starbops — APPROVED** [review]
LGTM, thank you.

**Comment by votdev** [comment]
To get this feature fully working the PR https://github.com/harvester/vm-import-controller/pull/66 is required as well.

**Pull request merged**